teps/0089-spire-support.md

```diff
@@ -0,0 +1,184 @@
+
+With the SPIRE integration, we successfully mitigate the two issues mentioned above.
+Tekton Chains can be confident the TaskRun was created by Tekton, since only Tekton workloads can have SPIRE issued SVIDs.
+Tekton Chains can also verify that the TaskRun yaml wasn't modified by anybody other than the Tekton controller, because the signature exists over the entire TaskRun yaml.
+
+_NOTE: A cluster administrator could still potentially threaten this model by disabling SPIRE or changing settings on the k8s Workload Attestor. At SLSA 3, it is still allowed that some trusted administrators will have root access to the build system. This will eventually need to be addressed for SLSA 4_.
+
+
+## Motivation
+
+Security! =)
+
+We'll also need this feature for Tekton to achieve [SLSA 3](https://slsa.dev/levels), which requires Non-falsifiable Provenance.
+
+### Requirements
+* SPIRE is configured to only issue SVIDs to the Tekton controller via a [k8s Workload Attestor](https://github.com/spiffe/spire/blob/main/doc/plugin_agent_workloadattestor_k8s.md)
+
+### Goals
+
+* Once a TaskRun completes, Pipelines can request a signature over the TaskRun yaml from SPIRE
+* Pipelines can store the signature and SVID on the TaskRun itself as annotaitons
+* Chains should be able to verify a TaskRun wasn't modified after execution via the signature and SVID provided
+
+### Non-Goals
+* Protect against a malicious cluster admin (this will be a goal for SLSA 4)
+
+## SPIRE Concepts
+The Tekton Controller will send the TaskRun yaml (this is the `payload`) to SPIRE for signing.
+SPIRE will make sure that the request came from the Tekton controller, and once it has confirmed that it will sign the `payload`.
+SPIRE will return the following to the Tekton Controller:
+1. The `signature` over the `payload`
+1. A SPIFFE Verifiable Identity Document, or [SVID](https://spiffe.io/docs/latest/spiffe-about/spiffe-concepts/#spiffe-verifiable-identity-document-svid), an X509 certificate containing the public key used to verify the `signature` against the `payload`
+
+## Proposal
+As mentioned above, the basic design looks like this:
+1. Tekton Pipelines receives a TaskRun config, and generates the Pod for it
+1. Tekton Pipelines runs the TaskRun, and then requests a signature and SVID over the completed TaskRun yaml from SPIRE
+1. Tekton Pipelines stores the SVID and signature as annotations on the TaskRun
+1. Tekton Chains observes the TaskRun, and verifies that SVID and signature against the TaskRun yaml
+1. If verification is successful, Chains proceeds normally. Otherwise, it stops and doesn't sign anything!
+
+
+Then, Chains gets the TaskRun.
+Chains will:
+1. Verify the signature against the SPIRE SVID
+1. If verification fails, Chains will mark the TaskRun as "failed" and move on
+1. Otherwise, continue signing stuff! 
+
+
+### Risks and Mitigations
+None that I can think of for now.
```

Comment:
is it possible to describe any additional dependencies the Tekton Pipelines controller will need to take on? e.g. libraries for interacting with SPIRE, signing, etc.

Reply:
added!